app/actions/service_instance_share.rb

```diff
@@ -30,7 +21,37 @@ def create(service_instance, target_spaces, user_audit_info)
 
     private
 
-    def supported_service_type?(service_instance)
+    def valid_target_spaces!(service_instance, target_spaces)
+      no_sharing_to_self!(service_instance, target_spaces)
+
+      target_spaces.each do |space|
+        plan_visibility!(service_instance, space)
+        name_uniqueness!(service_instance, space)
+      end
+    end
+
+    def plan_visibility!(service_instance, space)
+      visible_plans = ServicePlan.organization_visible(space.organization)
```

Comment:
We investigated and learnt that these two methods are functionally the same here. `space_visible` adds an additional filter for space-scoped brokers, but these can never be shared because by definition the plan is only available in the source space -- which makes that filter a no-op. We'll change this to use `space_visible` for improved readability anyway and add some test cases to be more explicit about space-scoped brokers.

Reply:
Changes for this included in #1033 and can be viewed here: https://github.com/cloudfoundry-incubator/cloud_controller_ng_sapi/commit/ff8276e5d201e573df03640a0f6865288b2d55af#diff-e3c96f58c4686b95721452ad906b78a9R42